File: go/1637.go
func maxWidthOfVerticalArea(points [][]int) int {
    xVals := make([]int, 0)
    for _,val := range points {
        xVals = append(xVals, val[0])
    }
    sort.Ints(xVals)
    diff := 0
    for i:=1; i < len(xVals); i++ {
        cur := xVals[i] - xVals[i-1]
        if cur > diff {
            diff = cur
        }
    }
    return diff
}

File: cpp/1637.cpp
class Solution {
public:
  int maxWidthOfVerticalArea(vector<vector<int>>& points) {
    vector<int> xVals;
    for (auto &point : points)
      xVals.push_back(point[0]);

    sort(xVals.begin(), xVals.end());

    int diff = 0;

    for (int i = 1; i < xVals.size(); i++) {
      int cur = xVals[i] - xVals[i-1];
      if (cur > diff)
        diff = cur;
    }
    return diff;
  }
};
